src/patterns/ec2-app.test.ts

```diff
@@ -248,18 +223,6 @@ describe("the GuEC2App pattern", function () {
       userData: "UserData from pattern declaration",
     });
 
-    expect(stack).toHaveResource("AWS::EC2::SecurityGroup", {
```

Comment:
This assertion feels unnecessary to me. This test is really checking that ingress rules can be overridden, which happens [here](https://github.com/guardian/cdk/pull/534/files#diff-39b32ce1aaa5782f5c37962a1acb30a4de4d7682adc29ea40b64bce8c8fe7d20L278-L288).

If we wanted to be really strict we could check that our custom ingress rule _doesn't_ yet exist at this point...

```
expect(stack).not.toHaveResource("AWS::EC2::SecurityGroup", {
  SecurityGroupIngress: [
    {
      CidrIp: "8.8.8.8/32",
      Description: "",
      FromPort: 1234,
      IpProtocol: "tcp",
      ToPort: 1234,
    },
  ],
});
```

...but I think this is overkill as the absurdity of this ingress rule means it's extremely unlikely that we'd ever add this by default!